npm/webpack-preprocessor/lib/cross-origin-callback-loader.ts

```diff
@@ -155,22 +157,20 @@ export default function (source: string, map, meta, store = crossOriginCallbackS
     },
   })
 
-  // if we found requires/imports, re-generate the code from the AST
-  if (hasDependencies) {
-    debug('callback with modified source')
-
-    // TODO: handle sourcemaps for this correctly
-    // https://github.com/cypress-io/cypress/issues/23365
-    // the following causes error "Cannot read property 'replace' of undefined"
-    //   return generate(ast, { sourceMaps: true }, source).code
-    // and can't pass in original map or the output ends up with
-    // `undefinedundefined` appended, which is a syntax error
-    this.callback(null, generate(ast, {}).code)
+  // if no requires/imports were found, callback with the original source/map
+  if (!hasDependencies) {
+    debug('callback with original source')
+    this.callback(null, source, map)
 
     return
   }
 
-  debug('callback with original source')
-  // if no requires/imports were found, callback with the original source/map
-  this.callback(null, source, map)
+  // if we found requires/imports, re-generate the code from the AST
+  const result = generate(ast, { sourceMaps: true }, {
+    [resourcePath]: source,
+  })
+  const newMap = mergeSourceMaps(map, result.map)
```

Comment:
What happens if you just return the new source map?

Reply:
The new source map is based off the code at that point in the webpack process. It's an intermediary version of the source. So if the original is TypeScript, it might through ts-loader, then babel, then get to this loader as some es5 version of the code without TypeScript. Since it's not the original source, resulting in stack traces and code frames being incorrect. The original source map needs to be merged with the one generated by the changes made in this loader for it the resulting one to be correct.

Reply:
@chrisbreiding minding adding some of this insight in as a comment for future devs? Sources maps are always tricky to finally get the hang of.

Reply:
Addressed in https://github.com/cypress-io/cypress/pull/24367/commits/78b97468b902f1687b72326924448d8bd1b7bc8a